rclcpp/include/rclcpp/event.hpp

```diff
@@ -17,20 +17,31 @@
 
 #include <atomic>
 #include <memory>
+#include <functional>
 
 #include "rclcpp/macros.hpp"
 #include "rclcpp/visibility_control.hpp"
 
 namespace rclcpp
 {
 
-class Event
+enum class ResourceStatusEvent
+{
+  DEADLINE_MISSED,
+  LIVELINESS_CHANGED,
+  LIFESPAN_EXPIRED
+};
+
+using ResourceStatusEventCallbackType = std::function<void(ResourceStatusEvent)>;
+
+
+class GraphEvent
```

Comment:
Why rename this? It is actually a generic event class, it doesn't have anything particular about it that is unique to graph events.

Reply:
Ah sorry, it looked like it was only ever used for node graph events (and it was otherwise only mentioned in the comments of `exceptions.hpp`). I guess sometime in the future, it will be used for something else. We'll revert this name change.

Reply:
Perhaps it's purpose/use should be combined with the purpose/use of `rcl_event_t`, I don't know, but in the case a name collision still happens, we could still consider renaming this. I just didn't see another `rclcpp::Event` yet.

Also if we rename it, we should move it to its own file, would have been `graph_event.hpp`. We try (but often fail) to live up to the "one class per file" ideal, where it makes sense of course.

Reply:
We will back out this change from this PR.